cmd/commands.go

```diff
@@ -0,0 +1,193 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/StackExchange/dnscontrol/models"
+	"github.com/urfave/cli"
+)
+
+// categories of commands
+const (
+	catMain  = "main"
+	catDebug = "debug"
+	catUtils = "utility"
+)
+
+// Run will execute the CLI
+func Run(version string) error {
+	app := cli.NewApp()
+	app.Version = version
+	app.Name = "dnscontrol"
+	app.HideVersion = true
+	app.Usage = "dnscontrol is a compiler and dsl for managing cloud dns zones"
+	app.Commands = []cli.Command{
+		*previewCommand,
+		*pushCommand,
+		*debugJSCommand,
+		*debugPreprocessCommand,
+		cli.Command{
+			Name:  "version",
+			Usage: "Print version information",
+			Action: func(c *cli.Context) {
+				fmt.Println(version)
+			},
+			Category: catDebug,
+		},
+	}
+	app.EnableBashCompletion = true
+	app.Run(os.Args)
+	return nil
+}
+
+// Shared config types
+
+// GetDNSConfigArgs contains what we need to get a valid dns config.
+// Could come from parsing js, or from stored json
+type GetDNSConfigArgs struct {
+	ExecuteDSLArgs
+	JSONFile string
+}
+
+func (args *GetDNSConfigArgs) flags() []cli.Flag {
+	return append(args.ExecuteDSLArgs.flags(),
+		cli.StringFlag{
+			Destination: &args.JSONFile,
+			Name:        "json",
```

Comment:
Suggestion: This flag should be "-input" or "-ir".  A flag named "-json" usually means to change the default format to JSON, which might be confusing.